### PR TITLE
Upgrade to PostgreSQL 15.1 via libpg_query 15-4.2.0

### PR DESCRIPTION
**Pull request by msepga**

### CODE REVIEW

**Comment by lfittl** [comment]
@msepga Awesome!

Looks like the two test failures are related to the source code line reported in the error message. I've noticed this in the past that this can be architecture/OS dependent. Maybe we need to remove the trailing `(...)` portion of the error message to avoid this problem.

**Comment by msepga** [comment]
Waiting on https://github.com/pganalyze/libpg_query/pull/170, to remove the [bespoke patches](https://github.com/pganalyze/pg_query/pull/270/files#diff-2220c646c9032856f35adf27a66fc7ca7c68372bdbbbd5ac7a5f96edc1a30cc3R125-R130) to match older Postgres data type layout

File: spec/lib/parse_spec.rb [hunk 1]
```diff
@@ -66,7 +60,7 @@
   it "handles errors" do
     expect { described_class.parse("SELECT 'ERR") }.to(raise_error do |error|
       expect(error).to be_a(PgQuery::ParseError)
-      expect(error.message).to eq "unterminated quoted string at or near \"'ERR\" (scan.l:1232)"
+      expect(error.message.gsub(/\ \(scan.l:\d+\)$/, '')).to eq "unterminated quoted string at or near \"'ERR\""
```
Comment (lfittl) [inline spec/lib/parse_spec.rb:63]: Nice, thanks for fixing this one :)

**Review by lfittl — APPROVED** [review]
Excellent - I think this looks good!

**Pull request merged**